src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
```

Comment:
PullProofAI

The `APIResponse.create` function can be improved by explicitly checking conditions for `data` and `error` to ensure clarity and consistency.
```diff
+ if (error) {
+   response.error = error;
+ } else if (!error && data) {
+   response.data = data;
+ } else {
+   response.data = {};
+ }
```


<!-- This is an auto-generated comment by PullProofAI -->